Controllers/controller_mpc.py

```diff
@@ -43,7 +44,18 @@ def configure(self, optimizer_name: Optional[str]=None):
             control_limits=self.control_limits,
             optimizer_logging=self.controller_logging,
             computation_library=self.computation_library,
-            **config_optimizer[optimizer_name],
+            **config_optimizer,
+        )
+        # Some optimizers require additional controller parameters (e.g. predictor_specification or dt) to be fully configured.
+        # Do this here. If the optimizer does not require any additional parameters, it will ignore them.
+        self.optimizer.configure(dt=self.config_controller["dt"], predictor_specification=predictor_specification)
+        
+        self.predictor.configure(
+            batch_size=self.optimizer.num_rollouts,
+            horizon=self.optimizer.mpc_horizon,
+            dt=self.config_controller["dt"],
+            computation_library=self.computation_library,
+            predictor_specification=predictor_specification
```

Comment:
- Added an optimizer configuration step
- Moved predictor configuration up to controller level. Before, this was done within the optimizer.